### ISSUE TITLE
Dask Cudf Join Fails when joining data with multiple partions

### ISSUE THREAD

**Issue by VibhuJawa** [rapidsai/dask-cudf#185]
# Dask Cudf join is Failing on Multiple Partitions 

Dask cudf merge is failing on large data with multiple partitions. 

It works on small data frames still trying to create a smaller reproduction code

##### Issue Gist view: 
Gist: https://gist.github.com/VibhuJawa/7947f1d87016c144cb4bc6f788f6ab21

**Comment by mrocklin** [comment]
cc @sklam in case you've run into this before

**Comment by sklam** [comment]
It would be good to know what dtypes are involved.  The usual reason for failure is mishandling null-values.

**Comment by VibhuJawa** [comment]
The join is being done on strings, I think this issue is similar to "Incorrect string to string join results when reading multiple CSV files in either df".  #187  .

**Comment by mrocklin** [comment]
Yes, I think that that issue probably has the information that @sklam is looking for.  It's often quite time consuming to run a full-workflow notebook to find an error, but looking at an example like what is in #187, that one can simply copy-paste into a terminal window, is much easier to understand what is going on without having to fully understand an application.

**Comment by VibhuJawa** [comment]
Understood, i will try harder to make smaller reproduction code in the future, i was trying but could not pin it down to the string join problem earlier.

**Comment by mrocklin** [comment]
Understood.  In this particular problem it wasn't a straightforward issue, so we requested a minimal example.  I think that this pattern that we did here makes sense going forwards:

-  Alice: Here is a general problem I'm having, any quick thoughts?
-  Bob: Nope, not in this case.  I recommend making a minimal example if you have the time
-  Alice: OK, here you go
-  Bob: OK, now I see what's going on ...

With the hope being for the following outcome sometimes:

-  Alice: Here is a general problem I'm having, any quick thoughts?
-  Bob: oh yeah, that's a known issue, see #XXX
